aws/resource_aws_guardduty_filter.go

```diff
@@ -0,0 +1,489 @@
+		case "less_than_or_equal":
+			value, err := conditionValueToInt(typedCriterion["values"].([]interface{}))
+			if err != nil {
+				return nil, fmt.Errorf("Value seems to be not an integer: %v", typedCriterion["values"].([]interface{})[0])
+			}
+			if criteria[typedCriterion["field"].(string)] == nil {
+				criteria[typedCriterion["field"].(string)] = &guardduty.Condition{
+					LessThanOrEqual: aws.Int64(value.(int64)),
+				}
+			} else {
+				criteria[typedCriterion["field"].(string)].LessThanOrEqual = aws.Int64(value.(int64))
+			}
+		case "not_equals":
+			criteria[typedCriterion["field"].(string)] = &guardduty.Condition{
+				NotEquals: aws.StringSlice(conditionValueToStrings(typedCriterion["values"].([]interface{}))),
+			}
+		}
+
+	}
+	log.Printf("[DEBUG] Creating FindingCriteria map: %#v", findingCriteria)
+
+	return &guardduty.FindingCriteria{Criterion: criteria}, nil
+}
+
+func conditionValueToStrings(untypedValues []interface{}) []string {
+	values := make([]string, len(untypedValues))
+	for i, v := range untypedValues {
+		values[i] = v.(string)
+	}
+	return values
+}
+
+func conditionValueToInt(untypedValues []interface{}) (interface{}, error) {
+	if len(untypedValues) != 1 {
+		return nil, fmt.Errorf("Exactly one value must be given for conditions like less_ or greater_than. Instead given: %v", untypedValues)
+	}
+
+	untypedValue := untypedValues[0]
+	typedValue, err := strconv.ParseInt(untypedValue.(string), 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("Parsing condition value failed: %s", err.Error())
+	}
+
+	return typedValue, nil
+}
+
+func parseImportedId(importedId string) (string, string, error) {
+	parts := strings.SplitN(importedId, "_", 2)
+	if len(parts) < 2 {
+		return "", "", fmt.Errorf("Error Importing aws_guardduty_filter record: '%s' Please make sure the record ID is in the form detectorId_name.", importedId)
+	}
+	return parts[0], parts[1], nil
+}
+
+func criterionResource() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"field": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"condition": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"values": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func flattenFindingCriteria(findingCriteriaRemote *guardduty.FindingCriteria) []interface{} {
+	var result []interface{}
+	var flatCriteria []interface{}
+
+	for field, conditions := range findingCriteriaRemote.Criterion {
+		if len(conditions.Equals) > 0 {
+			flatCriteria = append(flatCriteria, flattenStringCondition(field, "equals", conditions.Equals))
+		}
+		if len(conditions.NotEquals) > 0 {
+			flatCriteria = append(flatCriteria, flattenStringCondition(field, "not_equals", conditions.NotEquals))
+		}
+		if conditions.GreaterThan != nil {
+			flatCriteria = append(flatCriteria, flattenIntCondition(field, "greater_than", conditions.GreaterThan))
```

Comment:
The four `flattenIntCondition()` calls here could take an `int` parameter and have the dereference happen here
```suggestion
			flatCriteria = append(flatCriteria, flattenIntCondition(field, "greater_than", int(aws.Int64Value(conditions.GreaterThan))))
```

Reply:
https://github.com/suzuki-shunsuke/terraform-provider-aws/commit/59db7dd4624918589b69266026dd266bf44fa71a